connectors/s3-connector/src/main/scala/zio/connect/s3/S3Connector.scala

```diff
@@ -1,59 +1,85 @@
+    trace: Trace
+  ): ZSink[Any, S3Exception, CopyObject, CopyObject, Unit]
 
-  def copyObject(implicit trace: Trace): ZSink[Any, S3Exception, CopyObject, CopyObject, Unit]
+  def copyObject(
+    sourceRegion: => Region,
+    destinationRegion: => Region
+  )(implicit
+    trace: Trace
+  ): ZSink[Any, S3Exception, CopyObject, CopyObject, Unit]
 
-  def createBucket(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Unit]
+  def createBucket(
+    region: => Region
+  )(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Unit]
 
-  def deleteEmptyBucket(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Unit]
+  def deleteEmptyBucket(
+    region: => Region
+  )(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Unit]
 
-  def deleteObjects(bucketName: => BucketName)(implicit
+  def deleteObjects(bucketName: => BucketName, region: => Region)(implicit
     trace: Trace
   ): ZSink[Any, S3Exception, ObjectKey, ObjectKey, Unit]
 
-  final def existsBucket(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Boolean] =
+  final def existsBucket(
+    region: => Region
+  )(implicit trace: Trace): ZSink[Any, S3Exception, BucketName, BucketName, Boolean] =
     ZSink
       .take[BucketName](1)
       .map(_.headOption)
       .mapZIO {
         case Some(name) =>
           for {
-            listResponse <- listBuckets.runCollect
+            listResponse <- listBuckets(region).runCollect
             bucketExists  = listResponse.contains(name)
           } yield bucketExists
         case None => ZIO.succeed(false)
       }
 
   final def existsObject(
-    bucket: => BucketName
+    bucket: => BucketName,
+    region: => Region
   )(implicit trace: Trace): ZSink[Any, S3Exception, ObjectKey, ObjectKey, Boolean] =
     ZSink
       .take[ObjectKey](1)
       .map(_.headOption)
       .mapZIO {
         case Some(key) =>
           for {
-            listResponse <- listObjects(bucket).runCollect
+            listResponse <- listObjects(bucket, region).runCollect
             objectExists  = listResponse.contains(key)
           } yield objectExists
         case None => ZIO.succeed(false)
       }
 
-  def getObject(bucketName: => BucketName, key: => ObjectKey)(implicit trace: Trace): ZStream[Any, S3Exception, Byte]
+  def getObject(bucketName: => BucketName, key: => ObjectKey, region: => Region)(implicit
+    trace: Trace
+  ): ZStream[Any, S3Exception, Byte]
+
+  def listBuckets(region: => Region)(implicit trace: Trace): ZStream[Any, S3Exception, BucketName]
 
-  def listBuckets(implicit trace: Trace): ZStream[Any, S3Exception, BucketName]
+  def listObjects(bucketName: => BucketName, region: => Region)(implicit
+    trace: Trace
+  ): ZStream[Any, S3Exception, ObjectKey]
 
-  def listObjects(bucketName: => BucketName)(implicit trace: Trace): ZStream[Any, S3Exception, ObjectKey]
+  def moveObject(region: => Region)(implicit
```

Comment:
I think this can be implemented

```
final def moveObject(region: => Region)(implicit trace: Trace
   ): ZSink[Any, S3Exception, MoveObject, MoveObject, Unit] =
     moveObject(region, region)
```

Reply:
This change makes a lot of sense, thanks!